### DISCUSSION TITLE
How to create links that pass filter parameters to a table list?

### DISCUSSION THREAD

**Discussion by tstauch** [Budibase/budibase#4727] [Help]
I would like to link from a screen (e.g. User Profile) to a list screen, passing a filter as a URL parameter,
so that I can filter the list by certain conditions (eg user=uid)

What would be the right way to do this in Budibase?
I can't seem to find an example of this.

I have created a new screen, with a route that includes the filter field (e.g.

    .../filter_name/:filter_value

but this seems like a lot of work and requires maintenance of another list screen.
Is there a recommended way of doing this?

**Comment by mjashanks** [comment]
At the moment, this is the only way. We have talked about adding the ability to create "Reusable Component", whereby you create groups of components that you can save and use in multiple places.

For now, you can still use one screen. You just need to make sure that you are always passing URL parameters.

So, you could link to `/your_table/default` and then only apply the filter then `:filter_value is not default`.

**Reply by tstauch** [reply]
Ok, tks, I am nearly understanding the solution.

But please clarify, how would I implement what you mention:
   "only apply the filter then (sic, then?)  :filter_value is not default."

**Reply by mjashanks** [reply]
apologies for the late reply. So, first setup your data provider to filter based on your URL parameter

![image](https://user-images.githubusercontent.com/3524181/156595272-10c5e068-acd7-4940-8aab-832664c9671a.png)

Now, on the data provider, click "Configure Conditions", and set it up as follows:

![image](https://user-images.githubusercontent.com/3524181/156595876-d6adacc9-63ac-4d28-931c-a2ee1b006bfe.png)

You can click the "Define Filters" button to setup how your filter should act when the URL parameter is `default`.
